source/simulator/assembly.cc

```diff
@@ -1001,6 +399,9 @@ namespace aspect
 
     // Prepare the data structures for assembly
     scratch.finite_element_values.reinit (cell);
+    scratch.cell = cell;
+    scratch.pressure_scaling = material_model->reference_viscosity()
+                               / geometry_model->length_scale();
```

Comment:
As I mentioned earlier, it would be much better to use the simulator access for this instead.